### PR TITLE
[SMALLFIX] Improve debug logging in RpcUtils

### PR DESCRIPTION
**Pull request by apc999**
Originally the purpose of `RpcUtils.callAndLog` is to log the enter and exit info. However, `RpcUtils.callAndLog` has two issues:
- When log level is DEBUG, each exception on `FileSystemMaster` will be logged twice---once in `RpcUtils.call`, once in `RpcUtils.callAndLog` which is redundant.
- Even with redundant logging, the info is insufficient. Especially, tracestack of the exception is not shown in log. Simple fix by adding `e` in `logger.debug` in `callAndLog` will not work. This is because the exception is already turned into `TException` which has no tracestack info.

An example:
```
2017-11-19 20:59:45,555 DEBUG FileSystemMasterClientServiceHandler - Enter: ListStatus: path=/, options=ListStatusTOptions(loadDirectChildren:true, loadMetadataType:Once)
2017-11-19 20:59:45,560 DEBUG FileSystemMasterClientServiceHandler - ListStatus: path=/, options=ListStatusTOptions(loadDirectChildren:true, loadMetadataType:Once), Error=Authentication is not enabled
2017-11-19 20:59:45,564 DEBUG FileSystemMasterClientServiceHandler - Exit (Error): ListStatus: path=/, options=ListStatusTOptions(loadDirectChildren:true, loadMetadataType:Once), Error=Authentication is not enabled
```

This PR aims to address above two issues. After this PR, the above log message becomes:
```
2017-11-19 21:15:15,656 DEBUG FileSystemMasterClientServiceHandler - Enter: ListStatus: path=/, options=ListStatusTOptions(loadDirectChildren:true, loadMetadataType:Once)
2017-11-19 21:15:15,663 DEBUG FileSystemMasterClientServiceHandler - Exit (Error): ListStatus: path=/, options=ListStatusTOptions(loadDirectChildren:true, loadMetadataType:Once)
alluxio.exception.AccessControlException: Authentication is not enabled
        at alluxio.security.authentication.AuthenticatedClientUser.getClientUser(AuthenticatedClientUser.java:78)
        at alluxio.master.file.PermissionChecker.checkPermission(PermissionChecker.java:119)
        at alluxio.master.file.DefaultFileSystemMaster.listStatus(DefaultFileSystemMaster.java:803)
        at alluxio.master.file.FileSystemMasterClientServiceHandler$8.call(FileSystemMasterClientServiceHandler.java:250)
        at alluxio.master.file.FileSystemMasterClientServiceHandler$8.call(FileSystemMasterClientServiceHandler.java:245)
        at alluxio.RpcUtils.call(RpcUtils.java:39)
        at alluxio.master.file.FileSystemMasterClientServiceHandler.listStatus(FileSystemMasterClientServiceHandler.java:245)
        at alluxio.thrift.FileSystemMasterClientService$Processor$listStatus.getResult(FileSystemMasterClientService.java:1499)
        at alluxio.thrift.FileSystemMasterClientService$Processor$listStatus.getResult(FileSystemMasterClientService.java:1483)
        at org.apache.thrift.ProcessFunction.process(ProcessFunction.java:39)
        at org.apache.thrift.TBaseProcessor.process(TBaseProcessor.java:39)
        at alluxio.master.file.FileSystemMasterClientServiceProcessor.process(FileSystemMasterClientServiceProcessor.java:78)
        at org.apache.thrift.TMultiplexedProcessor.process(TMultiplexedProcessor.java:123)
        at org.apache.thrift.server.TThreadPoolServer$WorkerProcess.run(TThreadPoolServer.java:286)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```

### CODE REVIEW

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 1]
```diff
@@ -35,12 +35,23 @@
...
+        logger.warn("{}, Error={}", callable, e.getMessage());
+      }
       throw AlluxioStatusException.fromAlluxioException(e).toThrift();
     } catch (RuntimeException e) {
-      logger.error("{}", callable, e);
+      if (logger.isDebugEnabled()) {
+        logger.debug("Exit (Error): {}", callable, e);
+      } else {
+        logger.error("{}", callable, e);
+      }
```
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: simplify to `logger.error("Exit (Error): {}", callable, e)`?

File: core/server/common/src/main/java/alluxio/RpcUtils.java [hunk 2]
```diff
@@ -35,12 +35,23 @@
...
+      T ret = callable.call();
+      logger.debug("Exit (OK): {}", callable);
+      return ret;
     } catch (AlluxioException e) {
-      logger.debug("{}, Error={}", callable, e.getMessage());
+      if (logger.isDebugEnabled()) {
+        logger.debug("Exit (Error): {}", callable, e);
+      } else {
+        logger.warn("{}, Error={}", callable, e.getMessage());
+      }
```
Comment (calvinjia) [inline core/server/common/src/main/java/alluxio/RpcUtils.java]: I prefer
```
logger.debug("Exit (Error): {}", callable, e);
if (!logger.isDebugEnabled()) {
  logger.warn("{}, Error={}", callable, e.getMessage());
}
```

Comment (apc999) [reply core/server/common/src/main/java/alluxio/RpcUtils.java]: done

File: core/server/common/src/main/java/alluxio/RpcUtils.java (hunk 1 above)
Comment (apc999) [reply core/server/common/src/main/java/alluxio/RpcUtils.java]: done

**Comment by apc999** [comment]
@calvinjia PTAL

**Pull request merged**